### PR TITLE
allow scrolling props.parent instead of on window only

### PR DESCRIPTION
**Pull request by borisnadion**
usefull for https://github.com/seatgeek/react-infinite/
related to https://github.com/seatgeek/react-infinite/pull/47

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Cool! Curious what your use case is? Also why checkin the compiled code?

**Comment by KyleAMathews** [comment]
Hey would love to commit this. Please just remove the checked-in dist stuff.

**Comment by borisnadion** [comment]
Updated ;-) Not it's not changed.

**Comment by KyleAMathews** [comment]
Thanks! And if you could squash it down to one commit as well everything
looks great now.

**Comment by borisnadion** [comment]
done

**Comment by borisnadion** [comment]
parent has to be a function, we should bind a listener to an existing dom node

**Comment by KyleAMathews** [comment]
Why does parent have to be a function?

**Comment by KyleAMathews** [comment]
@borisnadion bumping this, would love to commit this. Why again does `parent` need to be a function? Why can't it just be the actual DOM node?

**Comment by borisnadion** [comment]
<ScrollableContent ref="scrollable" ...>
...      
</ScrollableContent>
<Headroom parent={-> @refs.scrollable?.getDOMNode() || window}>...</Headroom>

because the parent, or actually just source DOM element for measurements can actually be not parent at all, for example position=fixed elements inside scrollable ones look bad on iOS Safari: they're scrolled with a content and then jump to maintain fixed position when scroll is ended

**Comment by WickyNilliams** [comment]
Allows you to treat the element as a deferred value, i guess. 

Is there a danger of memory leaks due to not cleaning up your event handlers, if the element you listen to can change at any moment?

File: src/index.cjsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ module.exports = React.createClass
   ticking: false
 
   propTypes:
+    parent: React.PropTypes.node
```
Comment (KyleAMathews) [inline src/index.cjsx]: Parent should always be a function right? so `React.PropTypes.func`?

**Comment by KyleAMathews** [comment]
Cool :) makes sense. If you could rebase and address the question I raised, let's get this merged and in a new release.

**Comment by WickyNilliams** [comment]
Did you see my comment above regarding memory leaks and event listeners?

**Comment by KyleAMathews** [comment]
@WickyNilliams I did—not sure I understand—if the parent element is removed—wouldn't the event listeners get GCed?

**Comment by WickyNilliams** [comment]
Older IE will (React supports back to IE8). But I guess that's not such a priority these days haha. Stuck in my old ways :)

**Comment by viljamis** [comment]
Is there anything you need help with? I’d also need this feature in a project I’m working on. :-)

**Comment by KyleAMathews** [comment]
Released 1.7.0 w/ this PR included! Thanks @borisnadion!

**Pull request merged**